sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -124,6 +129,18 @@ public virtual bool IsClosed
             Logger.ClientCreateComplete(typeof(ServiceBusSender), Identifier);
         }
 
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusSender"/> class.
+        /// </summary>
+        /// <param name="entityPath">The entity path to send the message to.</param>
+        /// <param name="connection">The connection for the sender.</param>
+        internal ServiceBusSender(
```

Comment:
This may just be the diff looking odd, but did we add additional constructors?

Reply:
Yes I added a new constructor that takes in the options as well. I was getting API compat errors when just adding a default parameter.

Reply:
Interesting.   Internal constructors on a public type still flag the analyzer.   I guess there's no real harm in including it rather than suppressing the analyzer.     @JoshLove-msft:  Any concerns?

Reply:
Hmm, I'm not sure why Api Compat would care about internal constructors. I don't think it matters either way, but I am slightly curious as to what the error was.

Reply:
I believe the error was that the ServiceBusSender(path, connection) didn't exist in implementation but was in the contract

Reply:
oh... API Compat... I thought this was the analyzers.   Yeah, API Compat shouldn't care about the `internal` members.   But we have to be careful with `protected` ones.   If you want a hand looking at the API Compat thing, let me know.

Reply:
Ah yes, it makes sense that we'd need a new protected ctor, but should be able to reuse the existing internal one.